doc/build-osx.md

```diff
@@ -99,30 +99,22 @@ git clone https://github.com/bitcoin/bitcoin.git
 
 #### Wallet Dependencies
 
-It is not necessary to build wallet functionality to run `bitcoind` or  `bitcoin-qt`.
```

Comment:
IMO a comment like this is worth having still (here and in the other files), I'm not convinced the new format clearly conveys that the two types of wallet are both optional. Perhaps it would be good to start the wallet section off with a quick bit of info explaining the difference between the two subsections?

Reply:
I was hoping the fact that this exists in the section for optional dependencies would convey that both are optional. We could add back a line, but I'm trying to trim this down so it's not so overly verbose/repetitive.

Reply:
@fanquake fair enough, agreed!